src/lib/server/websearch/markdown/types.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable-next-line no-shadow */
+export enum MarkdownElementType {
+	Header = "HEADER",
+	Paragraph = "PARAGRAPH",
+	BlockQuote = "BLOCKQUOTE",
+	CodeBlock = "CODE_BLOCK",
+
+	UnorderedList = "UNORDERED_LIST",
+	OrderedList = "ORDERED_LIST",
+	UnorderedListItem = "UNORDERED_LIST_ITEM",
+	OrderedListItem = "ORDERED_LIST_ITEM",
+}
+
+type BaseMarkdownElement = {
```

Comment:
can it be generic like `BaseMarkdownElement<T = MarkdownElementType>` and then instead of `type ParagraphElement = BaseMarkdownElement & { type: MarkdownElementType.Paragraph };`, we can use it like `type ParagraphElement = BaseMarkdownElement<MarkdownElementType.Paragraph>`